numba_typing/arguments_annotation_default_func_class.py

```diff
@@ -0,0 +1,24 @@
+from inspect import signature, getfullargspec
+from typing import get_type_hints
+
+
+def get_arguments_annotation_default_func(method):
+    sig = signature(method)
+    argspec = getfullargspec(method)
+    arg_index = 0
+    annotations = {}
+    defaults = {}
+
+    for arg in argspec.args:
```

Comment:
Looks like you could use signature to iterate through parameters:
`for arg, in sig.parameters.values():`
Also let's fuse both for-loops to a single one.